ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextViewManager.java

```diff
@@ -80,6 +81,23 @@ public void setLineHeight(ReactTextView view, float lineHeight) {
     }
   }
 
+  // These values should match the values in ReactNativeStyleAttributes.js, where they're converted
+  // from JS values into the integer used in setTextDecorationLine.
+  private static final int UNDERLINE = 1 << 0;
+  private static final int LINE_THROUGH = 1 << 1;
+
+  @ReactProp(name = ViewProps.TEXT_DECORATION_LINE, customType = "TextDecorationLine")
+  public void setTextDecorationLine(ReactTextView view, int textDecorationLine) {
+    int paintFlags = view.getPaintFlags();
+    if ((textDecorationLine & UNDERLINE) == UNDERLINE) {
+      paintFlags |= Paint.UNDERLINE_TEXT_FLAG;
+    }
+    if ((textDecorationLine & LINE_THROUGH) == LINE_THROUGH) {
+      paintFlags |= Paint.STRIKE_THRU_TEXT_FLAG;
+    }
+    view.setPaintFlags(paintFlags);
```

Comment:
This is not correct. Consider the case when property is updated from "underline" to "line-through". In that case this method will be called first time with "underline" set in the mask and you'd update `paintFlags` to include `Paint.UNDERLINE_TEXT_FLAG`. Next when the updates comes in, this method will be called with `textDecorationLine` containing only "line-through" flag set. In that case you will set `Paint.STRIKE_THRU_TEXT_FLAG` to paintFlags but the `Paint.UNDERLINE_TEXT_FLAG` set previously will still be there.

There are many other cases when this will not work correctly including the one when due to update the "textDecorationLine" attribute has been removed from the component. Essentially what you should do here is to unset `Paint.UNDERLINE_TEXT_FLAG` and `Paint.STRIKE_THRU_TEXT_FLAG` in `paintFlags` before you start updating it